### PR TITLE
Update modulefiles on `rrfs_ci`

### PR DESCRIPTION
**Pull request by robgpita**
## DESCRIPTION OF CHANGES: 
* This PR refactors `modulefiles/build_aws_intel` to use `srw_common` to build & `modulefiles/wflow_aws` has been modified to correctly load rocoto and allow the miniconda3 installation to be discoverable. 
* In using `srw_common`, a couple of hpc-stack libraries were updated( [ec15cd7](https://github.com/ufs-community/ufs-srweather-app/commit/ec15cd71c884c0f09329e8a22b5cda441470e13d) ) on the self-hosted runner machine. 
* `build.yml` is removed, as the workflow fails (we do not maintain the `ncar` self hosted runner).  

## TESTS CONDUCTED: 
A Build test was successful on a self-hosted runner by attaching the `ci-aws-intel-build` label to this PR. Results of the GitHub Actions workflow can be found here: https://github.com/NOAA-GSL/ufs-srweather-app/actions/runs/2700522057.

## DEPENDENCIES:
None

## DOCUMENTATION:
This PR does not contribute any new changes that need to be documented. 

## CONTRIBUTORS (optional): 
@christinaholtNOAA @venitahagerty @christopherwharrop-noaa

### CODE REVIEW

**Comment by venitahagerty** [comment]
I looked through the code changes, and the Action steps. I'm curious as to the use of /scratch1. That's on AWS? 

Also, what does the "Check Build Sucess or failure step" do? I didn't find the output to be very descriptive.

**Comment by robgpita** [comment]
@venitahagerty Yes, `/scratch1` is the mount point for the Lustre FS on AWS (where miniconda and hpc-stack are installed, and where the End to End Tests are submitted from to use the compute nodes).  `Build_Status` is the final job which ultimately dictates the build status, displayed in the Github Actions UI. It is necessary to have this final job in the GHA workflow to check whether the build was successful or not, in checking the log emitted by `./build.sh`. Without the final `Build_Status` job, the build (and the GHA workflow) will always appear to succeed, because the previous job (`Build_SRWA`) only calls `./build.sh`, which does not exit on failure or success.

**Review by venitahagerty — APPROVED** [review]

**Review by christopherwharrop-noaa — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,21 +0,0 @@
-name: Build SRW
```
Comment (christinaholtNOAA) [inline .github/workflows/build.yml:1]: Do you have any ideas on strategies about how we might be able to manage GSL workflows alongside authoritative ones? (A question we can definitely dive into more in the wrap up)

**Review by christinaholtNOAA — APPROVED** [review]
This looks good. Just a broader scope question to think about below.

**Comment by christinaholtNOAA** [comment]
OBE.

**Pull request closed without merging**